src/writer.rs

```diff
@@ -0,0 +1,58 @@
+use std::io::{Write, Error, ErrorKind};
+use std::io;
+use InflateStream;
+
+pub struct InflateWriter<W: Write> {
+    inflater: InflateStream,
+    writer: W
+}
+
+impl<W: Write> InflateWriter<W> {
+    pub fn new(w: W) -> InflateWriter<W> {
+        InflateWriter { inflater: InflateStream::new(), writer: w }
+    }
+
+    pub fn finish(mut self) -> io::Result<W> {
```

Comment:
Oh cool, I didn't think of that. Then it seems like this patch, [flate2](https://github.com/alexcrichton/flate2-rs) and (https://github.com/oyvindln/deflate-rs) all are consistent, great!